### DISCUSSION TITLE
How to use template strings with graphql-codegen?

### DISCUSSION THREAD

**Discussion by joeprivettsmart** [dotansimha/graphql-code-generator#9128] [Q&A] [answered]
I have asked this question in StackOverflow...
https://stackoverflow.com/questions/75661685/how-to-use-template-strings-with-graphql-codegen

**Comment by n1ru4l** [comment, marked as answer]
You cannot use them as TypeScript cannot infer the type of that usage and codegen does not interpret your code.

**Reply by joeprivettsmart** [reply]
Not sure this makes sense "TypeScript cannot infer the type of that usage" - could you re-word it please? I would have thought in every case it would chose the string type (?)

**Reply by pm0u** [reply]
The type of your query is a string but there is no way at build time to determine what that query would return as this would require static analysis which is impossible if you introduce variables into the string that is your query. How could TS infer what fields your dynamic string is composed of?
